### DISCUSSION TITLE
Audio Playback: Best Practice Recommendation

### DISCUSSION THREAD

**Discussion by joshualambert** [appcelerator/titanium_mobile#12632] [Q&A]
I'm looking to implement an audio player to play a remote MP3 file. I'm looking for an audio player recommendation for handling:

1.) A progress bar that the user can drag around to move through the audio file.
2.) Buffering.
3.) Skip 10 seconds into future or past.

I've seen several examples laying around the internet, a lot of which seem to not have been updated in a while. Here's one of them: https://github.com/kosso/ti-audioplayer

Does anybody have a working sandbox, perhaps something built in Hyperloop, or something else entirely for doing a standard MP3 player app function? Open to all ideas. Please share code if you have it!

**Comment by macCesar** [comment]
It will also play live streams right ?? 👍🏼 

Can you also add:
- A way to prevent the OS from killing the App while in the background.
- Show basic controls from the locked screen like Spotify or YouTube music.

Thanks!!

**Reply by joshualambert** [reply]
Great point. Thanks for bringing that up also.

**Reply by macCesar** [reply]
You are welcome... I think they are essential functions!!

**Reply by macCesar** [reply]
This is very old.. I'm not even sure if still works... But it is iOS only 😔

https://github.com/hoyo/TiRemoteControl

**Comment by m1ga** [comment]
There is an example in the [hyperloop-examples repo](https://github.com/appcelerator/hyperloop-examples):
https://github.com/appcelerator/hyperloop-examples/blob/master/app/controllers/android/mediaplayer.js
that is using the same `AudioManager` that ti-audioplayer is using. So you could try to add the missing listeners there and use the HL version

**Comment by macCesar** [comment]
Hello Josh... How are you??

Any findings or tips on how to prevent the OS from killing the App while playing audio in the background?

I have an App ready to publish and I can't stop the OS from killing it after a few minutes in the background. 🤦‍♂️

Thanks in advance!!
